pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -403,6 +419,16 @@ public void reloadSegment(String segmentName, IndexLoadingConfig indexLoadingCon
     }
   }
 
+  public boolean canReuseExistingDirectoryForReload(SegmentZKMetadata segmentZKMetadata,
+      String currentSegmentTier, SegmentDirectory segmentDirectory, IndexLoadingConfig indexLoadingConfig,
+      Schema schema)
+      throws Exception {
+    boolean tierMigrationNeeded = indexLoadingConfig.getSegmentDirectoryLoader()
+        .equalsIgnoreCase(TierBasedSegmentDirectoryLoader.DIRECTORY_LOADER_NAME) && !StringUtils.equals(
+        segmentZKMetadata.getTier(), currentSegmentTier);
+    return !tierMigrationNeeded && !ImmutableSegmentLoader.needPreprocess(segmentDirectory, indexLoadingConfig, schema);
```

Comment:
curious why need to explicitly check if loader name is `tierBased`? With this check, it seems like other custom loaders that can do tier migration would be directed to reuse existing directory, when it's supposed to handle tier migration.